src/AlertBanner/AlertBanner.tsx

```diff
@@ -0,0 +1,102 @@
+/** @jsx jsx */
+import { jsx } from "@emotion/core";
+import React, { CSSProperties, useMemo } from "react";
+import PropTypes from "prop-types";
+
+import { base } from "../typography";
+import { colors } from "../colors";
+import { assertUnreachable } from "../shared/assertUnreachable";
+import { IconInfoSolid } from "../icons/IconInfoSolid";
+import { IconWarningSolid } from "../icons/IconWarningSolid";
+import { IconErrorSolid } from "../icons/IconErrorSolid";
+import { IconSuccessSolid } from "../icons/IconSuccessSolid";
+
+interface AlertBannerProps {
+  /**
+   * color theme for alert
+   * @default "light"
+   */
+  theme?: "light" | "dark";
+
+  /**
+   * The content of the banner, appears to right of icon
+   */
+  children?: React.ReactNode;
+
+  className?: string;
+  style?: CSSProperties;
+
+  /**
+   * Type of alert, this is used to determine the color and icon in the banner
+   */
+  type: "info" | "warn" | "error" | "success";
+}
+
+export const AlertBanner: React.FC<AlertBannerProps> = ({
+  children,
+  theme = "light",
+  type,
+  ...otherProps
+}) => {
+  const { Icon, color: colorTemp } = useMemo(() => {
+    switch (type) {
+      case "info":
+        return { color: colors.blue, Icon: IconInfoSolid };
+      case "warn":
+        return { color: colors.orange, Icon: IconWarningSolid };
+      case "error":
+        return { color: colors.red, Icon: IconErrorSolid };
+      case "success":
+        return { color: colors.green, Icon: IconSuccessSolid };
+      default:
+        assertUnreachable(type);
+    }
+  }, [type]);
```

Comment:
FYI: I originally didn't like this pattern, but it's grown on me as I've seen it more 👍